### ISSUE TITLE
[Python] DMatrix parameter `data_split_mode` is not documented

### ISSUE THREAD

**Issue by david-cortes** [dmlc/xgboost#9841]
Python class `DMatrix` allows a parameter `data_split_mode` in its constructor, but the docstrings leave this part blank where they should mention what it does: https://github.com/dmlc/xgboost/blob/7196c9d95e706566424ae0b7ea0046fe9c5a3849/python-package/xgboost/core.py#L838

**Comment by rongou** [comment]
Yeah got pulled into another project, will come back to this in a bit. @trivialfis feel free to assign this to me.
